### PR TITLE
Docker container to serve cloudefficiency out of ecs.

### PR DESCRIPTION
**Pull request by bytesandwich**

### CODE REVIEW

File: server2/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# generate self signed cert and run locally
```
Comment (JohnTheodore) [inline server2/README.md]: This is confusing, what does running all these comments get me (in addition to a self signed cert)

File: server2/publish_ssm_secrets.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# secrets
```
Comment (JohnTheodore) [inline server2/publish_ssm_secrets.sh]: add a shebang to this?

**Comment by JohnTheodore** [comment]
You need to update report/requirements.txt request

File: server2/selfsign/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM ubuntu:18.04
```
Comment (JohnTheodore) [inline server2/selfsign/Dockerfile]: Should there be a comment in here saying what this docker container does. also a docker run example, with the required environment variables, eg, $WEBSITE, $RANDOM, etc

File: server2/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Flask==1.0.2
```
Comment (JohnTheodore) [inline server2/requirements.txt]: If you're going to hard code requirements.txt (which makes sense so things don't break), you should probably have a travis.yml line for doing a security scans. it's too easy to forget about these and have a problem down the road.

**Comment by JohnTheodore** [comment]
What is 'server2', what was wrong with 'server', why do you need a second directory? are you just keeping two different systems in place, one for the crazy apiserver/lambda, and one for ecs. and later down the road you'll get rid of the apiserver/lambda one? it seems weird to keep both in the repo if you're not going to maintain one. please clarify

File: server2/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:18.04
```
Comment (JohnTheodore) [inline server2/Dockerfile]: please add a topline comment saying what this docker container will do in general, a high level summary

File: server2/app.py [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    token['iss'] = base_uri
+    token['aud'] = base_uri
+    token['exp'] = expires
+    token['iat'] = now
+    token['nbf'] = now
+    logger.debug('JWT params: %s', token)
+    return token
+
+
+class Cookie:
```
Comment (JohnTheodore) [inline server2/app.py]: add a comment, explaining what the cookie class is doing. ideally add a key/value set for what the actually cookie would look like in say the browser view

File: server2/app.py [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+def with_auth(func):
+    """
+    Authentication decorator.
+
+    Run the given request handler, passing it the user.
+    If the user has no valid auth token, then
+    redirect the user/resource owner to the keymaster
+    without invoking the given request handler.
+    """
+    @wraps(func)
```
Comment (JohnTheodore) [inline server2/app.py]: This is more aesthetic, but this decorator name is meh. perhaps importing it as something more reasonable, like 'authcheck' or something

File: server2/app.py [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+    def wrapped(*args, **kwargs):
```
Comment (JohnTheodore) [inline server2/app.py]: same idea. wrapped could be used as a name for any decorator. why not make the function name closer to it's functional purpose

File: server2/app.py [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+        **userinfo
+    )
+    cookie.add_jwt(oauth_token, jwt_token_params)
+    cookie.remove_jwt(oauth_state)
+    print('initial_path', initial_path)
+    return cookie.populate_resp(make_response(redirect(initial_path)))
+
+
+if __name__ == "__main__":
+    app.secret_key = os.urandom(24)
```
Comment (JohnTheodore) [inline server2/app.py]: Does this break things for users if you redeploy say twice in a day, and their session information stays the same?

Comment (bytesandwich) [reply server2/app.py]: This app doesn't use the session store, so I should just delete this. Good catch.

File: server2/README.md (hunk 1 above)
Comment (bytesandwich) [reply server2/README.md]: I added more documentation.

File: server2/publish_ssm_secrets.sh (hunk 2 above)
Comment (bytesandwich) [reply server2/publish_ssm_secrets.sh]: Done.

File: server2/selfsign/Dockerfile (hunk 3 above)
Comment (bytesandwich) [reply server2/selfsign/Dockerfile]: Done.

File: server2/requirements.txt (hunk 4 above)
Comment (bytesandwich) [reply server2/requirements.txt]: Done.

File: server2/Dockerfile (hunk 5 above)
Comment (bytesandwich) [reply server2/Dockerfile]: Done.

File: server2/app.py (hunk 6 above)
Comment (bytesandwich) [reply server2/app.py]: Done.

**Comment by bytesandwich** [comment]
I made these changes in https://github.com/Symantec/cloudefficiency/pull/9

**Pull request closed without merging**